src/librustdoc/doctest.rs

```diff
@@ -272,6 +327,12 @@ fn run_test(
     if as_test_harness {
         compiler.arg("--test");
     }
+    if options.json_unused_externs && !compile_fail {
```

Comment:
Why do we care if it is going to fail to compile? Do we not support printing unused externs in that case? (It seems reasonable to still, e.g.

```rust
extern crate hyper;

fn bar() { 
 let x: () = 5; // fails to compile, but we don't use hyper
}
```

Reply:
If something fails to compile, it might not fully run the stage of the compiler that would use the external crate. So hyper might *appear* to be used to folks but the compiler won't recognize it because of the compilation error. It might only see some crates, it might see no crates at all. All bets are off. So that's why I made compile fail cases not count into crate usages. If your external crate is only used in compile fail cases of rustdoc snippets, then maybe removing it is a good idea :).

Regarding your example, note that `extern crate hyper` marks the crate used by this lint just as any `pub use` or `use` reexport would :), see PR description.